axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/StockMoveServiceSupplychainImpl.java

```diff
@@ -268,6 +269,20 @@ public void cancel(StockMove stockMove) throws AxelorException {
     if (appSupplyChainService.getAppSupplychain().getManageStockReservation()) {
       reservedQtyService.updateReservedQuantity(stockMove, StockMoveRepository.STATUS_CANCELED);
     }
+    Boolean supplierArrivalCancellationAutomaticMail =
+        stockConfigService
+            .getStockConfig(stockMove.getCompany())
+            .getSupplierArrivalCancellationAutomaticMail();
+    if (!supplierArrivalCancellationAutomaticMail
+        || stockMove.getIsReversion()
+        || stockMove.getTypeSelect() != StockMoveRepository.TYPE_INCOMING) {
+      return;
+    }
+    Template supplierCancellationMessageTemplate =
+        stockConfigService
+            .getStockConfig(stockMove.getCompany())
+            .getSupplierArrivalCancellationMessageTemplate();
+    super.sendMailForStockMove(stockMove, supplierCancellationMessageTemplate);
```

Comment:
The mail sending should not be in the same transaction as the workflow, but be done after the first transaction. Please check how it is done for plan method:
* First, change the existing `cancel` method to `protected` and rename it to `cancelStockMove`.
* Create a new public `cancel` method, which needs to be NOT transactional. 
* Call the protected `cancelStockMove` method, check if we need to send a mail, then call `sendMailForStockMove`.

Separating transaction when sending mail is necessary to avoid issues preventing the mail to be sent on some cases where the process is too long.